src/libraries/System.Reflection.Emit/tests/AssemblyBuilderTests.cs

```diff
@@ -296,6 +296,31 @@ public void EqualsTest(AssemblyBuilder assembly, object obj, bool expected)
             }
         }
 
+        public class CustomAttribute : Attribute
+        {
+            public CustomAttribute()
+            {
+            }
+        }
+
+        [Fact]
+        public void GetReferencedAssemblies()
+        {
+            // create an assembly tagged with a custom attribute
+            var cattr_asm = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("custom_attr_assembly"), AssemblyBuilderAccess.Run);
+
+            ConstructorInfo classCtorInfo = typeof(CustomAttribute).GetConstructor(new Type[] { });
+            CustomAttributeBuilder cattr = new CustomAttributeBuilder(
+                        classCtorInfo,
+                        new object[] { });
+
+            Assert.Equal(0, cattr_asm.GetReferencedAssemblies().Length);
+
+            cattr_asm.SetCustomAttribute(cattr);
+
+            Assert.Equal(1, cattr_asm.GetReferencedAssemblies().Length); // Should now have a single reference, to this assembly
```

Comment:
It might be good to also check that `GetReferencedAssemblies()[0].Name` is the expected name.
The code below in `mono_assembly_get_assemblyref_checked` calls `mono_metadata_blob_heap_checked` and `mono_metadata_string_heap_checked` and it'd be nice to verify that that code is doing the right thing.

Reply:
Added check for `[0].Name` and also `[0].GetPublicKeyToken ()` which checks the `mono_metadata_blob_heap_checked` bit